packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -229,6 +234,10 @@ export async function processGraphQLRequest<TContext>(
 
       try {
         requestContext.document = parse(query, config.parseOptions);
+        extensionStack.setParsedDocumentAndOperationName(
```

Comment:
Yeah we can do that. I had originally thought that it would be a better solution! However, I wasn't willing to modify the type of the `EndHandler` returned from the `requestDidStart` function, since it breaks the types. With typescript inheritance/overrides, the functions need to be typed in the subclass as well, so I'd worry about breaking current implementations. 

https://github.com/apollographql/apollo-server/blob/4a2459e7308f47daf93f4e45c44637193f9e1afe/packages/graphql-extensions/src/index.ts#L26

Reply:
I don't think we need to necessarily change that type.  How do you feel about just using working within the scope of where we declare the `requestDidStart `'s `EndHandler` in `apollo-engine-reporting/../extension.ts`?

i.e. somewhere here:

https://github.com/apollographql/apollo-server/blob/7fb5a2914c23bf4df686d939cb9207a6cedf40f6/packages/apollo-engine-reporting/src/extension.ts#L232-L237

We could just add:

```ts
      const documentAST = this.documentAST || o.requestContext.document;
```

And then use `documentAST` rather than `this.documentAST` below it.